lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -254,89 +170,51 @@ export class UserAgentApplication {
-          logger = new Logger(null),
-          loadFrameTimeout = 6000,
-          navigateToLoginRequestUrl = true,
-          state = "",
-          isAngular = false,
-          unprotectedResources = new Array<string>(),
-          protectedResourceMap = new Map<string, Array<string>>(),
-          storeAuthStateInCookie = false
-      } = options;
-
-    // TODO: The configuration object will take care of many of these assignments
-    this.loadFrameTimeout = loadFrameTimeout;
-    this.clientId = clientId;
-    this.validateAuthority = validateAuthority;
-
-    // TODO: Replace string with constant
-    this.authority = authority || "https://login.microsoftonline.com/common";
-    this._navigateToLoginRequestUrl = navigateToLoginRequestUrl;
-    this._state = state;
-    this._isAngular = isAngular;
-    this._unprotectedResources = unprotectedResources;
-    this._protectedResourceMap = protectedResourceMap;
-    this._redirectUri = redirectUri;
-    this._postLogoutredirectUri = postLogoutRedirectUri;
-    this._logger = logger;
-    this.storeAuthStateInCookie = storeAuthStateInCookie;
-
-    // TODO: This should be replaced with two different callbacks for success and error cases
-    this._tokenReceivedCallback = tokenReceivedCallback;
-
-    // Track login and acquireToken in progress
-    this._loginInProgress = false;
-    this._acquireTokenInProgress = false;
-
-    // TODO: This should be replaced with cache object, typescript checking for "localStorage" and "sessionStorage" values
-    this._cacheLocation = cacheLocation;
-    if (!this._cacheLocations[cacheLocation]) {
-      throw new Error("Cache Location is not valid. Provided value:" + this._cacheLocation + ".Possible values are: " + this._cacheLocations.localStorage + ", " + this._cacheLocations.sessionStorage);
-    }
-    this._cacheStorage = new Storage(this._cacheLocation); //cache keys msal
+  constructor(configuration: Configuration, callback: tokenReceivedCallback) {
+
+    // Set the Configuration
+    this.config = configuration;
+
+    // Set the callback
+    this.tokenReceivedCallback = callback;
+
+
+    this.logger = this.config.system.logger;
+    this.clientId = this.config.auth.clientId;
+    this.inCookie = this.config.cache.storeAuthStateInCookie;
+
+    // if no authority is passed, set the default: "https://login.microsoftonline.com/common"
+    this.authority = this.config.auth.authority || DEFAULT_AUTHORITY;
+
+    // track login and acquireToken in progress
+    this.userLoginInProgress = false;
+    this.acquireTokenInProgress = false;
+
+    // cache keys msal - typescript throws an error if any value other than "localStorage" or "sessionStorage" is passed
+    try {
+        this.cacheStorage = new Storage(this.config.cache.cacheLocation);
+    } catch (e) {
+        this.config.system.logger.error("CacheLocation can be set only to 'localStorage' or 'sessionStorage' ");
```

Comment:
don't we set the local logger when config comes in?

Reply:
Yes. Need to refer the local logger here. Thanks for the catch.